src/Settings/ApplicationSettings/ApplicationSettingsScreen.tsx

```diff
@@ -1,85 +1,70 @@
 import React from 'react'
 import {defineMessages, useIntl} from 'react-intl'
-import {Platform, ScrollView, StyleSheet, Switch} from 'react-native'
+import {Alert, ScrollView, StyleSheet, Switch} from 'react-native'
 import DeviceInfo from 'react-native-device-info'
-import {useDispatch, useSelector} from 'react-redux'
 
+import {useAuthOsEnabledOnDevice, useAuthOsErrorDecoder, useAuthWithOs} from '../../auth'
 import {StatusBar} from '../../components'
-import {useCrashReports} from '../../hooks'
+import {useAuthMethod, useCrashReports, useRefetchOnFocus} from '../../hooks'
 import globalMessages from '../../i18n/global-messages'
-import {setAppSettingField} from '../../legacy/actions'
-import {APP_SETTINGS_KEYS} from '../../legacy/appSettings'
 import {CONFIG, isNightly} from '../../legacy/config'
-import {canBiometricEncryptionBeEnabled, isBiometricEncryptionHardwareSupported} from '../../legacy/deviceSettings'
-import KeyStore from '../../legacy/KeyStore'
-import {biometricHwSupportSelector, installationIdSelector, isSystemAuthEnabledSelector} from '../../legacy/selectors'
 import {isEmptyString} from '../../legacy/utils'
 import {useWalletNavigation} from '../../navigation'
+import {useStorage} from '../../Storage'
 import {useCurrencyContext} from '../Currency'
 import {NavigatedSettingsItem, SettingsBuildItem, SettingsItem, SettingsSection} from '../SettingsItems'
 
 const version = DeviceInfo.getVersion()
 
 export const ApplicationSettingsScreen = () => {
   const strings = useStrings()
-  const {navigation, navigateToSettings} = useWalletNavigation()
-  const isBiometricHardwareSupported = useSelector(biometricHwSupportSelector)
-  const isSystemAuthEnabled = useSelector(isSystemAuthEnabledSelector)
-  const installationId = useSelector(installationIdSelector)
-  const dispatch = useDispatch()
+  const {navigation} = useWalletNavigation()
+
   const {currency} = useCurrencyContext()
+  const storage = useStorage()
+  const {authMethod, refetch: refetchAuthMethod} = useAuthMethod(storage)
+
+  const {authOsEnabledOnDevice, refetch: refetchCanEnableOSAuth} = useAuthOsEnabledOnDevice({
+    // on emulator
```

Comment:
what is this for?

Reply:
testing without closing and opening the app on the emulator.

Reply:
thats more of an e2e test, isnt it?

Reply:
we should leave that to denis